lesson_4/part_2/shipyards.js

```diff
@@ -34,7 +34,7 @@ function ShipYard() {
     if (!(ship instanceof this.shipConstructor)) { return console.log("Type of ship is incorrect") };
 
     Object.assign(ship, new this.shipConstructor(...params));
-    ship.isOld = false;
+    Object.keys(ship.__proto__.__proto__).forEach(function(key) { console.log(key); delete ship[key] });
```

Comment:
Почему бы не сделать просто обход по собственным свойствам объекта корабля, т.к. мы его полностью заменяем новым ? 🙂
```suggestion
    Object.keys(ship).forEach(key => delete ship[key])
```